stackoverflow/src/main/java/ro/utcn/sd/cata/stackoverflow/controller/ConsoleController.java

```diff
@@ -0,0 +1,227 @@
+import ro.utcn.sd.cata.stackoverflow.service.UserService;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+@Component
+@RequiredArgsConstructor
+public class ConsoleController implements CommandLineRunner {
+    private final Scanner scanner = new Scanner(System.in);
+    private final UserService userService;
+    private final QuestionService questionService;
+    private final TagService tagService;
+    private final AnswerService answerService;
+    private boolean loggedIn = false;
+    private User loggedUser;
+
+    private boolean handleCommand(String command) {
+        switch (command) {
+            case "addUser":
+                handleAddUser();
+                return false;
+            case "removeUser":
+                handleRemoveUser();
+                return false;
+            case "login":
+                handleLogin();
+                return false;
+            case "addQuestion":
+                handleAddQuestion();
+                return false;
+            case "list":
+                handleList();
+                return  false;
+            case "findTag":
+                handleFindTag();
+                return false;
+            case "findTitle":
+                handleFindTitle();
+                return false;
+            case "addAnswer":
+                handleAddAnswer();
+                return false;
+            case "editAnswer":
+                handleEditAnswer();
+                return false;
+            case "removeAnswer":
+                handleRemoveAnswer();
+                return false;
+            case "exit":
+                return true;
+            default:
+                print("Unknown command. Try again.");
+                return false;
+        }
+    }
+
+    private void handleAddUser(){
+        print("Username:");
+        String username = scanner.next().trim();
+        print("Password:");
+        String password = scanner.next().trim();
+        User user = userService.addUser(username, password);
+        print("Created user: " + user + ".");
+    }
+
+    private void handleRemoveUser(){
+        print("User id:");
+        Integer id = scanner.nextInt();
+        userService.removeUser(id);
+        print("User removed.");
+    }
+
+    private void handleLogin(){
+        print("Enter username:");
+        String username = scanner.next().trim();
+        print("Enter password:");
+        String password = scanner.next().trim();
+        Optional<User> user = userService.findUserByUsernamePassword(username,password);
+        if(user.equals(Optional.empty())){
+            print("User or Password incorrect. Please try again");
+        }
+        else{
+            loggedUser =  user.get();
+            loggedIn = true;
+            print("Login successful. Welcome "+loggedUser.getUsername()+"! \n");
+        }
+    }
+
+    private void handleAddQuestion(){
+        if(loggedIn){
```

Comment:
You had to repeat this `loggedIn` check in each method --> Might be a better idea to do it somewhere centralized (for example in the `handleCommand` method.